File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F15.P2.go
package main

import (
	"cmp"
	"fmt"
)

// BEGIN PROBE F15.P2
func maxOf[T cmp.Ordered](a, b T) T {
	if a > b {
		return a
	}
	return b
}

func greater() int32 {
	m := maxOf[int32](3, 5)
	return m
}

// END PROBE F15.P2

func main() {
	fmt.Println(greater())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F15.P2.ts
function greater(): number {
// BEGIN PROBE F15.P2
function maxOf<T extends number | string>(a: T, b: T): T {
return a < b ? b : a
}
const m = maxOf(3, 5)
return m
// END PROBE F15.P2
}

console.log(greater())
